network/topology/dhtPeersTopology.go

```diff
@@ -0,0 +1,93 @@
+package topology
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	fcrypto "github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/network"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/p2p/dht"
+
+	discovery "github.com/libp2p/go-libp2p-discovery"
+)
+
+// DHTPeersTopology returns peers from inspecting the DHT
+type DHTPeersTopology struct {
+	fixedNodeID flow.Identity
+	routing     *discovery.RoutingDiscovery
+}
+
+func NewDHTPeersTopology(nodeID flow.Identity, routing *discovery.RoutingDiscovery) DHTPeersTopology {
+	return DHTPeersTopology{
+		fixedNodeID: nodeID,
+		routing:     routing,
+	}
+}
+
+func (r DHTPeersTopology) GenerateFanout(ids flow.IdentityList, _ network.ChannelList) (flow.IdentityList, error) {
```

Comment:
I was hoping to not use Topology and peer manager at all in the unstaked side of things. LibP2P `pubsub` accepts a `discovery` implementation and then [manages it for us](https://github.com/libp2p/go-libp2p-pubsub/blob/master/discovery.go) . So once we have our [DHT server setup](https://github.com/libp2p/go-libp2p-pubsub/blob/master/discovery.go#L8), we can pass it in as a pubsub options.

Our `Topology` and `PeerManager` we just not designed for such a thing.

Reply:
I'm happy with that. You can just cherry-pick the first commit, which just generates the discovery.